### ISSUE TITLE
Support Python3?

### ISSUE THREAD

**Issue by bananaappletw** [JonathanSalwan/Triton#521]
Triton is awesome project.
Thanks for the great work!!!

Is there any plan to support `python3`?
If that, maybe I could also help.
Thanks.

**Comment by JonathanSalwan** [comment]
I never tried but maybe it already supported? Did you tried?

**Comment by bananaappletw** [comment]
@JonathanSalwan 
Code seems like hard-coded python version as `2.7`.
Not providing way to customize python version
https://github.com/JonathanSalwan/Triton/blob/master/CMakeLists.txt#L115
https://github.com/JonathanSalwan/Triton/blob/master/src/libtriton/CMakeLists.txt#L282
Or something am I missing?
If I was wrong, correct me.
Thank you.

**Comment by JonathanSalwan** [comment]
Yeah it's hardcoded but if you remplace them does it work?

**Comment by JonathanSalwan** [comment]
If it works we can easily add an option to the compile process.

**Comment by bananaappletw** [comment]
@JonathanSalwan 
Seems not.
There are another places of code need to be modified.
![capture](https://cloud.githubusercontent.com/assets/6241350/25344844/f7a387a8-2945-11e7-84da-fad82ebc8e70.PNG)

**Comment by JonathanSalwan** [comment]
=(. Well, if you manage to figure out these issues I will be happy to accept your pull request :)

**Comment by bananaappletw** [comment]
@JonathanSalwan
Ok, I will try!
Thank you for kind reply.

**Comment by JonathanSalwan** [comment]
> Ok, I will try!

Woot, thanks. Go ahead from the `dev-next` branch.

**Comment by bananaappletw** [comment]
https://docs.python.org/3/howto/cporting.html
Seems like `python3` binding for c++ is completely different compared to `python2`.
It needs some work to do that.

**Comment by bananaappletw** [comment]
Quote from https://docs.python.org/2/c-api/intro.html
```
Note Since Python may define some pre-processor definitions which affect the standard headers on some systems, you must include Python.h before any standard headers are included.
```
The python binding codes should put `#include <Python.h>` before other headers

**Comment by RyanLee1119** [comment]
@JonathanSalwan It seems Python3 is a must for Windows is you want to use **Tigress_protection**, because the dependent **Arybo** cannot compile under Python 2.7 in Windows:

> with the exception of Python 2 not available under Windows 
https://pythonhosted.org/arybo/setup.html

Steps to reproduce in Windows:
1. Build Triton in Python2.7
2. In Tigress_protection run
`python ./solve-vm.py ./obfuscated_binaries/_binary_`
shows missing arybo.tools.triton_
3. pip install arybo shows error compling pytanque:
` building 'pytanque' extension
...
    include\pa/exprs.h(30) : fatal error C1083: Cannot open include file: 'cstdint': No such file or directory`

How can I bypass pytanque compilation in python2.7?
Thanks.

**Comment by JonathanSalwan** [comment]
You should open an issue on https://github.com/quarkslab/arybo

**Comment by JonathanSalwan** [comment]
Merged into `dev-v0.7`.
